### DISCUSSION TITLE
Montage bug - Having multiple images for 2x1 tiling, it decompose all images altogether, taking much time - please do it per couple images and not altogether.

### DISCUSSION THREAD

**Discussion by estatistics** [ImageMagick/ImageMagick#6126] [General]
You have 100 images that you like to make them into 2x1 tiles. Running montage, it decomposes all images together, and then make them in tiles. It may take more than 1 hour to finish while, if i execute montage specifying all "2x1" images specifically it may take less than 5 minutes. 

eg. `montage *.jpg -tile 2x1 -geometry +1+1  -quality 100 Tiles_No_.jpg (it makes Tiles_No_(0-1 or 0-2 etc.)) .jpg` 

So I propose, montage and other tools, to decompose / process images per requirement, and not altogether cause require much cpu, memory, hard disk space, and it is time consuming. :)

**Comment by estatistics** [comment]
![Uploading Επιλογή_106.jpg…]()

**Comment by fmw42** [comment]
You should write a script to loop over each pair of input images and process them with montage or just magick ... +append

You have a nice idea. However, the IM developers are very busy and cannot re-write montage to satisfy everyones different requirements. But Imagemagick is open source. So you or anyone else is free to write a different version and contribute it to Imagemagick.

**Reply by estatistics** [reply]
thanks! :)

**Comment by urban-warrior** [comment]
Add a size hint to your command, for example,

```
montage -define jpeg:size=128x128 "*.jpg[128x128]" -tile 2x1 -geometry +1+1  -quality 100 Tiles_No_.jpg 
```
